plugins/module_utils/network/sonic/facts/bgp_ext_communities/bgp_ext_communities.py

```diff
@@ -68,34 +68,38 @@ def get_bgp_extcommunities(self):
             match = member_config['match-set-options']
             permit_str = member_config.get('openconfig-bgp-policy-ext:action', None)
             members = member_config.get("ext-community-member", [])
-            result['name'] = name
+            result['name'] = str(name)
             result['match'] = match.lower()
-
+            result['members'] = dict()
+            result['type'] = 'standard'
+            result['permit'] = False
             if permit_str and permit_str == 'PERMIT':
                 result['permit'] = True
+            if members:
+                result['type'] = 'expanded' if 'REGEX' in members[0] else 'standard'
+            if result['type'] == 'expanded':
+                members = [':'.join(i.split(':')[1:]) for i in members]
```

Comment:
This instruction would be less confusing if it used a comma instead of a colon as the "join" character. The end result is the same regardless of what character is used. 

Is there some reason for using the colon here?

Reply:
In get_facts, we will receive members as "route-origin:301:301"
To strip "301:301" from this, used split(":") and considered only from 1 to n for joining.

Reply:
Okay. Thanks for explaining. This will work.

Another possible option that would remove the need to do the join and be slightly faster would be:

members = [(i.split(':', 1)[1]) for i in members]

Since the existing code works as required, I won't request this simplification.